project2-beat-that/script.js

```diff
@@ -1,4 +1,299 @@
+    }
+    else if (Number(playerTwoNumber) == Number(playerOneNumber)){
+      winner = `Draw`
+      output = `The result is a ${winner}. <br> Player 1 number is ${playerOneNumber}. <br>
+      Player 2 number is ${playerTwoNumber}. <br> `
+      
+      gameState = CONTINUE_GAME; 
+      return output;
+    }
+   
+  }
+
+  // CONTINUE GAME
+  if (gameState == CONTINUE_GAME){
+  counter = 0; 
+  while (counter < 2){
+    playerOne.pop();
+    playerTwo.pop();
+    counter +=1; 
+  }
+  gameState = ROLL_DICE;
+    output = `Press submit to continue playing! <br> `; 
+    return output;  
+  }
+
+// CHECK SCORE  
+  if (gameState == SCORE){
+    console.log(`Player 1 combined arrary is ${playerOneCombine}`)
+    var index = 0; 
+    let player1TotalScore = 0; 
+    let player2TotalScore = 0; 
+    for (index = 0; index < playerOneCombine.length; index +=1){
+      player1TotalScore = player1TotalScore + Number(playerOneCombine[index]); 
+      // console.log(`Player 1 total score = ${player1TotalScore}`);   
+  }
+  console.log(`Player 2 combined arrary is ${playerTwoCombine}`)
+  for (index = 0; index < playerTwoCombine.length; index +=1){  
+    player2TotalScore = player2TotalScore + Number(playerTwoCombine[index]); 
+    // console.log(`Player 2 total score = ${player2TotalScore}`); 
+  }
+  if (player1TotalScore == 0 ||player2TotalScore == 0 ){
+    gameState = CONTINUE_GAME; 
+
+    return `KEEP ON PLAYING`;
+  }
+  else if (player1TotalScore < player2TotalScore){
+    winner = `Player 2`; 
+  }
+  else if (player1TotalScore > player2TotalScore){
+    winner = `Player 1`; 
+  }
+  
+  output = `===   LEADERBOARD   === <br>
+  The current leader is ${winner}. <br> Player 1 has a score of ${player1TotalScore} <br> Player 2 has a score of ${player2TotalScore} <br> 
+  Press submit to continue playing`;
+  gameState = CONTINUE_GAME; 
+  return output; 
+
+  }
+
+  // Lowest Combined Number Mode
+  if (gameState = LOWEST_COMBINE_NUMBDER) {
+  
+    var index = 0; 
+    let player1TotalScore = 0; 
+    let player2TotalScore = 0; 
+    for (index = 0; index < playerOneCombine.length; index +=1){
+      player1TotalScore = player1TotalScore + Number(playerOneCombine[index]); 
+    
+    for (index = 0; index < playerTwoCombine.length; index +=1){  
+      player2TotalScore = player2TotalScore + Number(playerTwoCombine[index]); 
+    }
+
+    if (player1TotalScore == 0 ||player2TotalScore == 0 ){
+      gameState = CONTINUE_GAME; 
+  
+      return `KEEP ON PLAYING`;
+    }
+    else if (player1TotalScore < player2TotalScore){
+      winner = `Player 1`; 
+    }
+    else if (player1TotalScore > player2TotalScore){
+      winner = `Player 2`; 
+    }
+    output = `===   LEADERBOARD (Reversed)   === <br>
+    The current leader is ${winner}. <br> Player 1 has a score of ${player1TotalScore} <br> Player 2 has a score of ${player2TotalScore} <br> Press submit to continue playing`;
+  gameState = CONTINUE_GAME; 
+  return output; 
+}
```

Comment:
again repeated logic with your `SCORE` game mode, see how you can make this a common helper function that outputs different leaderboard based on whether its in highest or lowest combine number mode